BT-Tracking/Interface/Risky Encounters/PreviousRiskyEncountersVC.swift

```diff
@@ -9,31 +9,84 @@
-    private lazy var dateFormatter: DateFormatter = {
-        let dateFormatter = DateFormatter()
-        dateFormatter.timeStyle = .none
-        dateFormatter.dateStyle = .medium
-        return dateFormatter
-    }()
     private let disposeBag = DisposeBag()
+    private var dataSource: RxTableViewSectionedReloadDataSource<PreviousRiskyEncountersVM.Section>!
+
+    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        dataSource = nil
+
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+
+        setupDataSource()
+    }
+
+    required init?(coder: NSCoder) {
+        dataSource = nil
+
+        super.init(coder: coder)
+
+        setupDataSource()
+    }
 
     override func viewDidLoad() {
         super.viewDidLoad()
 
         title = viewModel.title
 
-        viewModel.previousExposures.bind(to: tableView.rx.items) { [dateFormatter] tv, _, element in
-            let cell = tv.dequeueReusableCell(withIdentifier: "PreviousRiskyEncountersCell") ?? UITableViewCell()
-            cell.textLabel?.text = dateFormatter.string(from: element.date)
-            cell.selectionStyle = .none
-            return cell
-        }.disposed(by: disposeBag)
+        viewModel.sections
+            .bind(to: tableView.rx.items(dataSource: dataSource))
+            .disposed(by: disposeBag)
+
+        tableView.rx
+            .modelSelected(Exposure.self)
+            .subscribe(onNext: { [weak self] value in
+                if let indexPath = self?.tableView.indexPathForSelectedRow {
+                    self?.tableView.deselectRow(at: indexPath, animated: true)
+                }
+                self?.perform(segue: StoryboardSegue.RiskyEncounters.showDetail, sender: value)
+            })
+            .disposed(by: disposeBag)
 
         tableView.tableFooterView = UIView()
     }
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        super.prepare(for: segue, sender: sender)
+
+        switch StoryboardSegue.RiskyEncounters(segue) {
+        case .showDetail:
+            let controller = segue.destination as? RiskyEncountersDetailVC
+            controller?.exposure = sender as? Exposure
+        default:
+            break
+        }
+    }
+
+    private func setupDataSource() {
+        dataSource = RxTableViewSectionedReloadDataSource<PreviousRiskyEncountersVM.Section>(configureCell: { [weak self] _, _, _, item in
+            self?.configureCell(item) ?? UITableViewCell()
+        })
+        dataSource.titleForHeaderInSection = { dataSource, index in
+            dataSource.sectionModels[index].model
+        }
+    }
+
+    private func configureCell(_ item: Exposure) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "PreviousRiskyEncountersCell") ?? UITableViewCell()
+        cell.textLabel?.text = self.viewModel.dateFormatter.string(from: item.date)
+        #if DEBUG || !PROD
```

Comment:
Nejsem si jisty, jestli !PROD funguje. Pokud nemas definovany primo flag !PROD, tak bych cekal, ze to nikdy neprojde.

Reply:
Funguje to, zkousel jsem to :)